.environment/terraform-fmt/run-terraform-fmt.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+function error() {
+    echo "terraform-fmt> ERROR: ${*}"
+}
+
+function warning() {
+    echo "terraform-fmt> Warning: ${*}"
+}
+
+function note() {
+    echo "terraform-fmt> info: ${*}"
+}
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+TERRAFORM_NEEDS_FMT=0
+LOGFILE="terraform-fmt.log"
+
+function terraform_check_fmt() {
+    note "Checking Terraform formatting."
+    make -C "${REPO_ROOT?}/operations" -f "${REPO_ROOT?}/operations/Makefile" tf-cmd TF_CMD="terraform fmt -check -recursive /app/src" > "${REPO_ROOT?}/${LOGFILE?}" 2>&1
+    TERRAFORM_NEEDS_FMT=$?
+    return ${TERRAFORM_NEEDS_FMT?}
+}
+
+function terraform_fmt() {
+    if [[ ${TERRAFORM_NEEDS_FMT?} != 0 ]]; then
+        warning "Formatting all Terraform files."
+        make -C "${REPO_ROOT?}/operations" -f "${REPO_ROOT?}/operations/Makefile" tf-cmd TF_CMD="terraform fmt -recursive /app/src" >> "${REPO_ROOT?}/${LOGFILE?}" 2>&1
+        return $?
+    fi
+    return 0
+}
+
+terraform_check_fmt
+RC=$?
+
+terraform_fmt
```

Comment:
Would it make more sense to condition execution of `terraform_fmt` on the value of `${RC?}` (and then _not_ do the check inside `terraform_fmt`, thus making it callable from anywhere without having to set the `${TERRAFORM_NEEDS_FMT}=1`)  or is there another reason why `terraform_fmt` is the one doing the check?

```suggestion
if [[ ${RC?} != 0 ]]; then
    terraform_fmt
fi
```

Reply:
Refactored the code so this is no longer an issue.